### ISSUE TITLE
Divergence after 18000 Iterations.

### ISSUE THREAD

**Issue by haohao900618** [chuckcho/video-caffe#11]
I run the example of "example/c3d_ucf101/train_ucf101.sh", it goes well at begining and it get top-1 accuracy 0.24 at 18000 Iterations. But after 18000 it diverges suddenly, the top-1 accuracy goes to 0.008. I don't know what's the matter with it. How can I solve the problem. And anyone else have encountered the problem like me.
thanks for you!

.......
Iteration 18000, Testing net (#0)
Test net output #0: accuracy/top-1 = 0.24
Test net output #1: accuracy/top-5 = 0.486
Test net output #2: loss = 3.30735 (* 1 = 3.30735 loss)
Iteration 19000, Testing net (#0)
Test net output #0: accuracy/top-1 = 0.008
Test net output #1: accuracy/top-5 = 0.037
Test net output #2: loss = 87.3366 (* 1 = 87.3366 loss)
........

**Comment by chuckcho** [comment]
Were you able to solve this? Maybe, by decreasing the base learning rate?

**Comment by robertmash2** [comment]
I ran into a similar issue with an Alexnet derived classifier while augmenting training data with random polygonal occlusions.  I tried reducing the learning rate drastically, but never solved it.  It'd start at loss=2.3 and train down to loss=~0.75 and then, in one learning step the loss would return to the starting value of ~2.3 and then stay there.  It didn't 'start over' and begin learning again, it'd just sit there at ~2.3 until I killed the session.  I'm inclined to think that I made the state space topology difficult to traverse in some way by randomly occluding portions of the training images.  Anyone else have similar experiences?

**Comment by chuckcho** [comment]
I debugged the code, and it should be able to converge and match the performance as in the original paper. See https://github.com/chuckcho/video-caffe/blob/master/examples/c3d_ucf101/README.md for detail.

**Comment by ParitoshParmar** [comment]
It is still giving me a loss of 87.3366 as @haohao900618 . Loss decreases, but almost suddenly rises to 87.3366. Any insights ? I am using the updated code.

**Comment by robertmash2** [comment]
My problem ended up being related to input normalization.  I forgot to scale the augmented training data to the range of 0-1,resulted in training divergence.  After properly scaling the images down to a consistent range, the solver worked as normal.  
The lesson that I took away, is that normalization is critical to convergence.  In future, if I see divergent training behavior I'll first think about whether I've done anything to affect normalization.

**Comment by ParitoshParmar** [comment]
@robertmash2 Thanks for the suggestion, I will try it out.

**Comment by cteckwee** [comment]
I encountered divergence after 5000 iterations with this updated codes. I notice that the train_test prototxt input doesn't scale to 0-1 (i.e., there is no scale: 0.00390625 in transform_param). Could this be the reason? Btw, I am using NVidia K6000 12GB, Cuda 7.5 and cudnn v4.

**Comment by cteckwee** [comment]
I just reduced the base learning rate to 0.001 instead of 0.003 and decrease the batch size from 30 to 15, then the training went well. Test accuracy hovers around 45% when reaching 5th epoch. 

It is weird that I couldn't train with larger batch size (for e.g. 30 will always trigger cuda_status error) even though my GPU card has 12GB of RAM. Never encountered this with older video-caffe version. I'm running on ported Windows version though.


![ucf101trainoutput](https://cloud.githubusercontent.com/assets/8372411/17431429/886183e6-5b2c-11e6-9db4-f32d136a6616.png)

**Comment by chuckcho** [comment]
@cteckwee thanks for sharing. `batch_size` issue is mysterious. i'll let you know if i find something.

**Comment by ParitoshParmar** [comment]
I increased the effective batch size to 90 by using iter_size: 3 and like @cteckwee ,  I got a divergence after initial convergence. I also normalized the input as suggested by @robertmash2 .

**Comment by chuckcho** [comment]
I made a fix on video reading routine -- https://github.com/chuckcho/video-caffe/commit/a6c5c1b626817f368520a343e837963823242194 I hope this fixes the funny batch_size business you guys are seeing.
